ci-operator/templates/openshift/installer/cluster-launch-installer-remote-libvirt-e2e.yaml

```diff
@@ -0,0 +1,594 @@
+            if [[ -n "${libvirt_network}" ]]; then
+                cat > /tmp/artifacts/installer/worker-hostrecords.xml << EOF
+        <host ip='192.168.126.51'>
+          <hostname>alertmanager-main-openshift-monitoring.apps.${NAMESPACE}-${JOB_NAME_HASH}.${BASE_DOMAIN}</hostname>
+          <hostname>console-openshift-console.apps.${NAMESPACE}-${JOB_NAME_HASH}.${BASE_DOMAIN}</hostname>
+          <hostname>downloads-openshift-console.apps.${NAMESPACE}-${JOB_NAME_HASH}.${BASE_DOMAIN}</hostname>
+          <hostname>grafana-openshift-monitoring.apps.${NAMESPACE}-${JOB_NAME_HASH}.${BASE_DOMAIN}</hostname>
+          <hostname>oauth-openshift.apps.${NAMESPACE}-${JOB_NAME_HASH}.${BASE_DOMAIN}</hostname>
+          <hostname>prometheus-k8s-openshift-monitoring.apps.${NAMESPACE}-${JOB_NAME_HASH}.${BASE_DOMAIN}</hostname>
+        </host>
+        EOF
+                echo "Libvirt network found. Injecting worker DNS records."
+                mock-nss.sh virsh --connect "${REMOTE_LIBVIRT_URI}" net-update --network "${libvirt_network}" --command add-last --section dns-host --xml "$(< /tmp/artifacts/installer/worker-hostrecords.xml)"
+                break
+            fi
+        done
+
+        wait "${openshift_install}"
+
+    # Performs cleanup of all created resources
+    - name: teardown
+      image: "${IMAGE_LIBVIRT_INSTALLER}"
+      terminationMessagePolicy: FallbackToLogsOnError
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-profile
+        mountPath: /etc/openshift-installer
+      - name: artifacts
+        mountPath: /tmp/artifacts
+      env:
+      - name: INSTANCE_PREFIX
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      - name: KUBECONFIG
+        value: /tmp/artifacts/installer/auth/kubeconfig
+      - name: USER
+        value: test
+      - name: HOME
+        value: /tmp
+      - name: NSS_WRAPPER_PASSWD
+        value: /tmp/test-passwd
+      - name: NSS_WRAPPER_GROUP
+        value: /tmp/test-group
+      - name: NSS_USERNAME
+        value: test
+      - name: NSS_GROUPNAME
+        value: test
+      command:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        function queue() {
+          local TARGET="${1}"
+          shift
+          local LIVE="$(jobs | wc -l)"
+          while [[ "${LIVE}" -ge 45 ]]; do
+            sleep 1
+            LIVE="$(jobs | wc -l)"
+          done
+          echo "${@}"
+          if [[ -n "${FILTER}" ]]; then
+            "${@}" | "${FILTER}" >"${TARGET}" &
+          else
+            "${@}" >"${TARGET}" &
+          fi
+        }
+
+        function teardown() {
+          set +e
+          touch /tmp/shared/exit
+          export PATH=$PATH:/tmp/shared
+
+          echo "Gathering artifacts ..."
+          mkdir -p /tmp/artifacts/pods /tmp/artifacts/nodes /tmp/artifacts/metrics /tmp/artifacts/bootstrap /tmp/artifacts/network
+
+
+          #if [ -f /tmp/artifacts/installer/terraform.tfstate ]
```

Comment:
Should we uncomment this? Do the other jobs still collect logs from the bootstrap machine?

Reply:
I don't remember why we commented this out originally, but https://github.com/openshift/release/pull/6386/files#diff-328023b6106bef905766021160e63d5dR478 suggests to me that the test container would need access to the bootstrap machine - or a jump node to get there. As I recall, the AWS e2e suite defines an additional "bastion" node that it can use to jump to the bootstrap node for logs.

Reply:
We do forward traffic to the bootstrap machine in the environment though. That's how I was debugging some of the earlier failures.

Reply:
Looking at this in greater detail, I think that to get this to work, we'd need to allow SSH traffic into the LPAR. This is something we are working towards removing, so I will go ahead and clear it for now. If we determine that we really desire bootstrap logs, we can look into creating a mechanism for doing so.